java/ql/lib/semmle/code/java/JDK.qll

```diff
@@ -198,39 +197,6 @@ class TypeFile extends Class {
 }
 
 // --- Standard methods ---
-/**
- * Any constructor of class `java.lang.ProcessBuilder`.
- */
-class ProcessBuilderConstructor extends Constructor, ExecCallable {
-  ProcessBuilderConstructor() { this.getDeclaringType() instanceof TypeProcessBuilder }
-
-  override int getAnExecutedArgument() { result = 0 }
-}
-
-/**
- * Any of the methods named `command` on class `java.lang.ProcessBuilder`.
- */
-class MethodProcessBuilderCommand extends Method, ExecCallable {
-  MethodProcessBuilderCommand() {
-    this.hasName("command") and
-    this.getDeclaringType() instanceof TypeProcessBuilder
-  }
-
-  override int getAnExecutedArgument() { result = 0 }
-}
-
-/**
- * Any method named `exec` on class `java.lang.Runtime`.
- */
-class MethodRuntimeExec extends Method, ExecCallable {
```

Comment:
I only deprecated things extending `ExecCallable`, mostly to avoid deprecation errors in our CI.

This kind of classes are useful as a sorts of "public, general use" CodeQL API that models common objects in the JDK, so even if they aren't used anywhere I _think_ they don't hurt. But again, I can deprecate them if you have a strong opinion about it.

Reply:
> This kind of classes are useful as a sorts of "public, general use" CodeQL API that models common objects in the JDK, so even if they aren't used anywhere I think they don't hurt.

This makes sense to me.
I also don't have a strong opinion regarding this. Was mainly asking for my own understanding if I'm ever deprecating something similar in the future. 🙂